### DISCUSSION TITLE
Vim support

### DISCUSSION THREAD

**Discussion by nikivdev** [notaapp/talk#88] [Feature Requests]
I'd love to be able to use vim bindings to move around.

**Comment by astoilkov** [comment]
Makes sense. Do you currently use another app that has vim bindings?

**Comment by nikivdev** [comment]
Yes, I use [Sublime Text](https://www.sublimetext.com) with with [Six Vim Plugin](https://github.com/guillermooo/Six) as my note taking app.

**Comment by astoilkov** [comment]
Cool. Thanks!

Is Vim support blocking you from experimenting with Nota more?

Do you have an extra plugin in Sublime Text adding functionality related to note-taking?

I hope you don't mind asking questions. 🙏

**Comment by nikivdev** [comment]
> Is Vim support blocking you from experimenting with Nota more?

Yes it's a big blocker.

> Do you have an extra plugin in Sublime Text adding functionality related to note-taking?

No, I do have [Alfred plugin to search notes and few little things](https://github.com/nikitavoloboev/small-workflows#personal-workflows) thus Nota is potentially a very interesting replacement to my setup.

But yes, vim plugin is an issue for me. My most common actions are things like `di(` or `di[` to delete inside () or []. As well as deleting until a character like `dt]` to delete until next `]`. Or things like `ciw` to cut a word.

**Comment by astoilkov** [comment]
Thanks! I will post any updates we have on this here.

**Comment by led** [comment]
+1 for vim support

**Comment by manuraj17** [comment]
+1 for vim support

I use vim mode in most of my daily apps. This will cement my usage of nota more.

**Comment by dvcrn** [comment]
Another vote for vim support!

**Comment by naoric** [comment]
Something I would definitely pay for

**Comment by nikivdev** [comment]
No vim support is the only reason I didn't migrate to Nota from Obisian/SublimeText I use now. Also plugins.

**Comment by slaughtr** [comment]
Yeah turns out that without vim keybindings I am more or less useless at text editing, I'm too used to them. Not even sure I can give Nota a proper evaluation because it's such a huge difference compared to my normal workflows.

**Comment by dvmrry** [comment]
Same boat here - a vi(m) command line extension would be absolutely fantastic

**Comment by jcha0713** [comment]
The lack of vim key bindings in the Nota app is preventing me from using it. Any updates on this?

**Reply by astoilkov** [reply]
Hey, we see the significant interest in this feature.

We don't have the expertise on this (as both of us haven't worked with Vim) but we are planning to take a look at how others are doing it. Which apps do this feature well and which apps don't? What should we look out for? Can we miss something important? Teach us.

**Comment by luvs** [comment]
+1

**Comment by elamaranae** [comment]
+1 for this feature

Perhaps the ability to use neovim instance would be great like this plugin does for vscode
https://github.com/vscode-neovim/vscode-neovim

**Comment by Auratelience** [comment]
+1

**Comment by led** [comment]
As maran-e states, building on neovim would be the only sane way to achieve this. It's a modern, complete (and then some) implementation of vim. Implementing your own vim functionality would be a nightmare drain on your resources.

**Comment by saccarosium** [comment]
+1 for the feature. The biggest problem I have with the application.
